fvm/crypto/crypto.go

```diff
@@ -150,6 +150,22 @@ func CryptoToRuntimeHashingAlgorithm(h hash.HashingAlgorithm) runtime.HashAlgori
 	}
 }
 
+// ValidatePublicKey returns true if public key is valid
+func ValidatePublicKey(signAlgo runtime.SignatureAlgorithm, pk []byte) (valid bool, err error) {
+	sigAlgo := RuntimeToCryptoSigningAlgorithm(signAlgo)
+
+	_, err = crypto.DecodePublicKey(sigAlgo, pk)
+
+	if err != nil {
+		// TODO: `crypto.IsInvalidInputsError(err)` doesn't work here!
+		if _, ok := err.(*crypto.InvalidInputsError); ok {
+			return false, nil
+		}
+		return false, err
```

Comment:
maybe we could wrap the error:
```suggestion
		return false, fmt.Errorf("validate public key failed: %w", err)
```